website/docs/faqs/Warehouse/db-connection-dbt-compile.md

```diff
@@ -0,0 +1,17 @@
+---
+title: Why does dbt compile need a data platform connection?
+description: "`dbt compile` needs a data platform connection to perform introspective queries and generate the compiled SQL"
+sidebar_label: "Why does dbt compile need a data platform connection?"
+id: db-connection-dbt-compile
+---
+
+The [`dbt compile`](reference/commands/compile) command is a really important step in the dbt workflow. This is because it transforms the raw SQL in dbt models and tests into compiled SQL, which is appropriate to run on a specific data platform. The compiled SQL includes SQL queries that are generated by dbt, such as query parts for incremental models, `ref` macros, and other [dbt-specific syntax](reference/node-selection/syntax).
```

Comment:
>really important step in the dbt workflow

This is true, but a common misconception is that it's necessary to `dbt compile` before you `dbt run`, and I worry that this reinforces that misconception. I think you could drop that whole sentence and not do any harm:

The [`dbt compile`](reference/commands/compile) command ~is a really important step in the dbt workflow. This is because it~ transforms the raw SQL in dbt models

Reply:
>The compiled SQL includes SQL queries that are generated by dbt, such as query parts for incremental models, `ref` macros, and other [dbt-specific syntax](reference/node-selection/syntax).

I'm not super clear on what this means

Reply:
> This is true, but a common misconception is that it's necessary to `dbt compile` before you `dbt run`, and I worry that this reinforces that misconception.

big +1 to this